### DISCUSSION TITLE
Delete a node between 2 nodes

### DISCUSSION THREAD

**Discussion by syntrydy** [xyflow/xyflow#2845] [Q&A] [answered]
Imagine we have 3 nodes A->B->C connected to each other, On deleting the node B, we need to have A & C connected automatically by a new edge (A->C). 

Currently I have a delete node by Id method, something like

```js
const deleteNodeById = (id) => {
    flowInstance.setNodes((nds) => nds.filter((node) => node.id !== id))
 } 
```

how can I use it in this context? Thanks !

**Comment by CamboimGabriel** [comment, marked as answer]
You can try to detect if there's an edge that uses that id as target and another one as source... if this is true you can proceed by getting the source of the first edge and the target of the second one and pushs a new edge...

Here is an example of the conditional

```
const sourceEdge = edges.find(edge => edge.source === id)
const targetEdge = edges.find(edge => edge.target === id)

if( sourceEdge && targetEdge ){
        flowInstance.setEdges((edges) => [...edges, {id: "randomIdOfYourChoice", target: sourceEdge.target, source: targetEdge.source}])
}
```

That's only an example, but should work in that way inside your deleteNodeById!

**Reply by syntrydy** [reply]
Thanks @CamboimGabriel, I took a similar approach that works
- first I have a method` getTargetIdOfDeletedNode(nodeId, flowInstance) `to get the target id
- followed by `createNewEdge(flowInstance, data)` which creates a new edge between the parent node of the deleted node and our target node

After the node is deleted and the two other nodes reconnected, I noticed an empty gap, left by the deleted node. What could be the best approach to recompute the nodes position ?

**Reply by CamboimGabriel** [reply]
you can just calculate the deleted node position + width/height and change the position of the target node!!
